bitnami/kafka/templates/_helpers.tpl

```diff
@@ -346,19 +353,457 @@
+  {{- end -}}
+  {{ printf "%s" (join "," $res)}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns the controller quorum voters based on the number of controller-eligible nodes
+*/}}
+{{- define "kafka.extraListeners.containerPorts" -}}
+{{- range $listener := .Values.listeners.extraListeners -}}
+- name: {{ lower $listener.name}}
+  containerPort: {{ $listener.port }}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns the zookeeper.connect setting value
+*/}}
+{{- define "kafka.zookeeperConnect" -}}
+{{- if .Values.zookeeper.enabled -}}
+{{- printf "%s:%s%s" (include "kafka.zookeeper.fullname" .) (ternary "3181" "2181" .Values.tls.zookeeper.enabled) (tpl .Values.zookeeperChrootPath .) -}}
+{{- else -}}
+{{- printf "%s%s" (join "," .Values.externalZookeeper.servers) (tpl .Values.zookeeperChrootPath .) -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns the controller quorum voters based on the number of controller-eligible nodes
+*/}}
+{{- define "kafka.kraft.controllerQuorumVoters" -}}
+{{- if .Values.kraft.controllerQuorumVoters -}}
+  {{ printf "%s" (tpl .Values.kraft.controllerQuorumVoters $) }}
+{{- else -}}
+  {{- $controllerVoters := list -}}
+  {{- $fullname := include "common.names.fullname" . -}}
+  {{- $releaseNamespace := include "common.names.namespace" . -}}
+  {{- range $i := until (int .Values.controller.replicaCount) -}}
+  {{- $nodeId := add (int $i) (int $.Values.controller.minId) -}}
+  {{- $nodeAddress := printf "%s-controller-%d.%s-controller-headless.%s.svc.%s:%d" $fullname (int $i) $fullname $releaseNamespace $.Values.clusterDomain (int $.Values.service.ports.controller) -}}
+  {{- $controllerVoters = append $controllerVoters (printf "%d@%s" $nodeId $nodeAddress ) -}}
+  {{- end -}}
+  {{- join "," $controllerVoters -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Section of the server.properties configmap shared by both controller-eligible and broker nodes
+*/}}
+{{- define "kafka.commonConfig" -}}
+log.dir={{ printf "%s/data" .Values.controller.persistence.mountPath }}
+{{- if or (include "kafka.saslEnabled" .) }}
+sasl.enabled.mechanisms={{ upper .Values.sasl.enabledMechanisms }}
+{{- end }}
+# Interbroker configuration
+inter.broker.listener.name={{ .Values.listeners.interbroker.name }}
+{{- if regexFind "SASL" (upper .Values.listeners.interbroker.protocol) }}
+sasl.mechanism.inter.broker.protocol={{ upper .Values.sasl.interBrokerMechanism }}
+{{- end }}
+{{- if (include "kafka.sslEnabled" .) }}
+# TLS configuration
+ssl.keystore.type=JKS
+ssl.truststore.type=JKS
+ssl.keystore.location=/opt/bitnami/kafka/config/certs/kafka.keystore.jks
+ssl.truststore.location=/opt/bitnami/kafka/config/certs/kafka.truststore.jks
+#ssl.keystore.password=
+#ssl.truststore.password=
+#ssl.key.password=
+ssl.client.auth={{ .Values.tls.sslClientAuth }}
+ssl.endpoint.identification.algorithm={{ .Values.tls.endpointIdentificationAlgorithm }}
+{{- end }}
+{{- if .Values.acl.authorizerClassName }}
+authorizer.class.name={{ .Values.acl.authorizerClassName }}
+{{- end }}
```

Comment:
I see values like `acl.allowEveryineIfNoAclFound` that are directly translated into a `allow.everyone.if.no.acl.found` entry. In other charts like Milvus or APISIX, we have something like this

```
  defaultConfig: |
    {{- if .Values.dataPlane.metrics.enabled }}
    plugin_attr:
      prometheus:
        export_uri: /apisix/prometheus/metrics
        metric_prefix: apisix_
        enable_export_server: true
        export_addr:
          ip: 0.0.0.0
          port: {{ .Values.dataPlane.containerPorts.metrics }}
    {{- end }}
    apisix:
      node_listen: {{ .Values.dataPlane.containerPorts.http }}

  extraConfig: {}
```
(the reason for having the default configuration as a value is for giving more flexibility to advanced users to modify all configurations at their will, even though that may imply other parts of the deployment not working as expected. This was taken from other upstream charts examples like grafana-tempo)

Suppose that I want to change `plugin_attr.prometheus.metric_prefix`, I could do something like this in my custom values.yaml file:

```
extraConfig:
  plugin_attr:
    prometheus:
      metric_prefix: test_
```

or to do `helm install ... --set extraConfig.plugin_attr.prometheus.metric_prefix=test_`. This is afterward merged and overridden with the proper priority order in an init container. Would it be possible to do something similar here? It would make the chart simpler and more flexible

Reply:
My issue with ACL is that we are setting default `allowEveryineIfNoAclFound=true`:
https://github.com/bitnami/charts/blob/3bfcc0812171abc52f11edc32c2d31650a1bbc8c/bitnami/kafka/values.yaml#L223-L225

While Kafka sets it default to `false` (safest) and discourages users to set it to `true` in production environments:
https://docs.confluent.io/platform/current/kafka/authorization.html#super-users

For that reason, I made it so it relies on default settings unless explicitly set.

Reply:
The thing is, would it be possible to extend the extra config section to allow overriding values?

Reply:
I'm not sure what you mean by 'overriding values'.
Configuration can be extended using the new `extraConfig` value or entirely override using `config`/`existingConfigmap`. 
Overriding may be complicated as I don't know how Kafka handles duplicated settings.

That is why, by default, the configuration would be minimal:
- Kraft settings (if kraft enabled): `node.id, process.roles, controller.quorum.voters , controller.listener`
- Zookeeper settings (if zookeeper enabled): `broker.id, zookeeper.connect`
- Listeners settings: `listeners, listeners.security.protocol.map, advertised.listeners`
- Sasl settings: `listener.name.*.jaas.config`
- TLS settings: `(certs location, passwords, etc.) + listener.name.*.ssl.client.auth`

Using a default value for `acl.*` settings would mean including them in the list of 'default' settings

Reply:
What about the `mergeOverwrite` function that we use in other charts?

Reply:
I removed `acl.*` parameters in favor of `extraConfig`